cfme/cloud/instance/__init__.py

```diff
@@ -404,25 +453,46 @@ def step(self, *args, **kwargs):
 
     def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
-        if self.view.entities.adv_search_clear.is_displayed:
+        if self.view.adv_search_clear.is_displayed:
             logger.debug('Clearing advanced search filter')
-            self.view.entities.adv_search_clear.click()
+            self.view.adv_search_clear.click()
         self.view.toolbar.reload.click()
 
 
 @navigator.register(Instance, 'AddFloatingIP')
 class AddFloatingIP(CFMENavigateStep):
-    prerequisite = NavigateToSibling('Details')
+    VIEW = AddFloatingIPView
+
+    def prerequisite(self, *args, **kwargs):
+        if not self.obj.provider.one_of(OpenStackProvider):
+            raise DestinationNotFound('Add Floating IP only available for Openstack instances')
+        return navigate_to(self.obj, 'Details')
 
     def step(self):
-        if version.current_version() >= '5.7':
             try:
                 self.prerequisite_view.toolbar.configuration.item_select(
                     'Associate a Floating IP with this Instance')
             except NoSuchElementException:
                 raise DestinationNotFound('Add Floating IP option not available for instance')
-        else:
-            raise DestinationNotFound('Floating IP assignment not available for appliance version')
+
+
+@navigator.register(Instance, 'AttachVolume')
+class AttachVolume(CFMENavigateStep):
+    VIEW = AttachVolumeView
+
+    def prerequisite(self, *args, **kwargs):
+        if not self.obj.provider.one_of(OpenStackProvider):
```

Comment:
I'll let this in for now, but in future, I want this functionality to be on the provider class, so we can query it there instead of making a conditional inside the core.